docs/arch/monitor.md

```diff
@@ -0,0 +1,93 @@
+# Monitor
+
+## Contributors
+
+- Sarah THEOULLE
+- Pauline CONTAT
+- Thomas BROINE
+- Baptiste BRONSIN
+
+## Functionalities
+
+- Listening to events from a remote Git repository
+- Recognizing the event type
+- Adapting actions according to the event type and then calling the controller via an external API
+
+## What
+
+Based on user provided configuration, the monitor listens for specific events from a remote Git repository and takes actions based on them. We need to recognize two types of events: `Commit` and `Pull Request`. Depending on the event type, an HTTP request will be sent to the controller.
+
+`POST` /pipeline :
+
+**Body**:
+
+- `name`: A `string` that corresponds to the pipeline name.
+- `body`: A `file` that is generated by the monitor and contains the actions to be executed by the controller.
+
+>[!Note]
+> The request **will** be a multipart/form-data since the pipeline file could be quite long.
+
+## Why
+
+The goal is to trigger the controller to launch a CI process according to the detected event from the remote repository.
+
+## How
+
+**Set Up the Git Repository:**
+In the CLI, you can launch the monitoring while giving the following parameters :
+
+- `--config`: The path to the config file
+- `--url`: The url of the git repository to watch
+- `--event`: The type of event to listen to
+- `--pipeline`: The path to the actions file
+
+If you provide the `--config` argument, the other options are not mandatory.
+
```

Comment:
I think the CLI should have its own documentation.  
Where will go the config datas ?

Reply:
The CLI is the main part of our feature, it does not make sense to create another documentation for it.
And I did not understand your second comment

Reply:
Your CLI is going to talk to a component, the monitor. You configure the monitor through the CLI, in order for it to monitor events and eventually send requests to the controller. So there is an interface between your CLI and the monitor. Is it HTTP based? gRPC?

Reply:
We thought the monitor will be configured by the server CLI on the first launch.

Reply:
> Our CLI will communicate with the monitor using an HTTP-based interface.

Right. That interface should be described and documented here.

Reply:
> We thought the monitor will be configured by the server CLI on the first launch.

What do you mean by server CLI?